### ISSUE TITLE
CS3/CS4: Patch Request

### ISSUE THREAD

**Issue by zephyroths** [AdmiralCurtiss/SenPatcher#126]
is it possible to request patches to fix issue on Linux/Proton or Steam Deck where the game freezes if you want to quit the game normally?

**Comment by AdmiralCurtiss** [comment]
Yes, that would be nice, though I admit I'm not sure where to start there with debugging it.

**Comment by zephyroths** [comment]
will a log of it helps? 
[Uploading steam-991270.log…]()

**Comment by AdmiralCurtiss** [comment]
Yeah, CS3 also randomly freezes when quitting on Windows. I think it's some threading issue during the shutdown sequence where a subsystem shuts down before some other code still tries to use it (IIRC you also see a bunch of error messages about waiting on invalid handles in the debug output) but I haven't actually investigated it in detail.

**Comment by AdmiralCurtiss** [comment]
(fwiw the brutal solution would be to just capture the exit request and redirect it to std::quick_exit but that seems pretty ugly)
